imports.py

```diff
@@ -0,0 +1,203 @@
+	return x.replace(f'{delim}slash', '/')
+
+
+def url_and_params_to_filename(url: str, params: Dict) -> str:
+	"""Convert to filename"""
+	x = url
+	if params:
+		x += '___params_'
+		for k, v in params.items():
+			x += '__' + k + '_' + v
+	filename = encode_filename_str(x) + '.json'
+	return filename
+
+
+def cache_write(url: str, params: Dict, response: Dict):
+	"""Write to cache"""
+	filename = url_and_params_to_filename(url, params)
+	if not os.path.exists(CACHE_DIR):
+		os.makedirs(CACHE_DIR)
+	path = os.path.join(CACHE_DIR, filename)
+	with open(path, 'w') as f:
+		json.dump(response, f, indent=4)
+
+
+def cache_read(url: str, params: Dict) -> Union[Dict, None]:
+	"""Read from cache"""
+	filename = url_and_params_to_filename(url, params)
+	path = os.path.join(CACHE_DIR, filename)
+	if os.path.exists(path):
+		with open(path, 'r') as f:
+			data: Dict = json.load(f)
+		return data
+	return None
+
+
+# Proper funcs / classes
+def decode_response_status_not_ok(response: Response) -> bool:
+	"""Decode status"""
+	if response is None:
+		print("NO RESPONSE")
+		return True
+	else:
+		if 200 <= response.status_code < 300:
+			return False
+		elif 400 <= response.status_code < 500:
+			return True
+		elif response.status_code >= 500:
+			return True
+
+
+def decode_code_system_json(response: Dict):
+	"""Decode CodeSystem response"""
+	# if not decode_response_status_not_ok(response):
+	# 	response_dict = json.loads(response.text)
+	if 'entry' in response:
+		print("title: " + response['entry'][0]['resource']['title'])
+		print("resourceType: ", response['entry'][0]['resource']['resourceType'], end="  ")
+		print("id: " + response['entry'][0]['resource']['id'])
+		print("url: " + response['entry'][0]['resource']['url'])
+		print("full url: " + response['entry'][0]['fullUrl'])
+		id_idx = 1 if len(response['entry']) > 1 else 0
+		id_sys = response['entry'][id_idx]['resource']['identifier'][0]['system']
+		id_sys_val =  response['entry'][id_idx]['resource']['identifier'][0]['value']
+		print("identifier system: \"" + id_sys, end="\",  ")
+		print("value: \"" + id_sys_val + "\"")
+	else:
+		print("Not Present")
+
+
+def decode_validate_response(response, optional_prints=False):
+	"""Decode validate response"""
+	print(response)
+	if optional_prints:
+		print(response.keys())
+		print(response['issue']['severity'])
+		print(response['issue']['response'])
+		display(HTML(response['text']['div']))
+		print("")
+
+
+def decode_lookup_response(response: Dict):
+	"""Deocde lookup response"""
+	# if not decode_response_status_not_ok(response):
+	# 	response_dict = json.loads(response.text)
+	if 'issue' in response:
+		print("error")
+	else:
+		for p in response['parameter']:
+			if p['name'] != 'property' and p['name'] != 'designation':
+				if 'valueString' in p:
+					print(p['name'], ":", p['valueString'], sep="", end=", ")
+				if 'valueBoolean' in p:
+					print(p['name'], ":", p['valueBoolean'], sep="", end=", ")
+		print("")
+
+
+class TimsClient:
```

Comment:
Created `TimsClient` class. Can instantiate like this:
```python
client = TimsClient(base_url=BASE_URL, use_cache=True)
```

`use_cache` is an optional param and defaults to `False`.